rules_building_block/persistence_suspicious_file_opened_through_editor.toml

```diff
@@ -0,0 +1,108 @@
+[metadata]
+creation_date = "2023/07/25"
+maturity = "production"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
+updated_date = "2023/07/25"
+integration = ["endpoint"]
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule monitors for the potential edit of a suspicious file. In Linux, when editing a file through an editor, a 
+temporary .swp file is created. By monitoring for the creation of this .swp file, we can detect potential file edits of
+suspicious files. The execution of this rule is not a clear sign of the file being edited, as just opening the file 
+through an editor will trigger this event. Attackers may alter any of the files added in this rule to establish
+persistence, escalate privileges or perform reconnaisance on the system. 
+"""
+from = "now-119m"
+interval = "60m"
+index = ["logs-endpoint.events.*", "endgame-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Suspicious File Edit"
+risk_score = 21
+rule_id = "3728c08d-9b70-456b-b6b8-007c7d246128"
+severity = "low"
+tags = ["Domain: Endpoint", "OS: Linux", "Use Case: Threat Detection", "Tactic: Persistence", "Tactic: Privilege Escalation", "Data Source: Elastic Endgame", "Rule Type: BBR"]
+timestamp_override = "event.ingested"
+type = "eql"
+building_block_type = "default"
+
+query = '''
+file where event.action in ("creation", "file_create_event") and file.extension == "swp" and 
```

Comment:
Why are we not using change or file modify events such as 
- change 
- updated 
- renamed or rename will also file name changes for sensitive files.

We already have a similar rule to monitor /etc/ld.so.preload which was tuned in 8.9 Telemetry analysis and event action was added. Where we found updated event was very helpful in reducing false positives as it reflected the changes to the file. rules/linux/privilege_escalation_ld_preload_shared_object_modif.toml

I am a bit unclear why .swp files? 

Am also providing a sample file event actions that could be of help 
creation
rename
renamed
changed-file-permissions-of
attributes_modified
changed-file-ownership-of
created
moved

Reply:
https://github.com/elastic/detection-rules/pull/2960#discussion_r1273555352

Reply:
The edit file should also be reflected in "changed" or "updated" right? Make it enough to monitor the original file as opposed to .swp files?

Reply:
@Aegrah I am curious to check if you run the suggested query with change or updated for original file and not the swp file extension ones do we still see the same result in ur test cluster like you have hits for the swp files.
If not then it gets interesting to understand what does changed or updated actually reflect in file events

Reply:
@shashank-elastic The whole point of this rule is to bypass our current limitations with regards to file edits. We are currently not able to detect file edits in any other way. As everything is a file, Linux always needs to keep a current state and a new state while editing in an editor. That's why .swp files are created when editing files using an editor, so it can afterwards overwrite the new file with the .swp file, or just remove the .swp file if no changes are made. The logic provided in this rule bypasses our limitation, by looking at the .swp files in order to detect file edits, because of the .swp file creation event.

Reply:
@shashank-elastic https://github.com/elastic/detection-rules/issues/1608